### PR TITLE
Add functionality to generate random password

### PR DESCRIPTION
**Pull request by isaack-njama**
This PR addresses #4364, generating a random password for now if one is not provided to encrypt sensitive config files

### CODE REVIEW

File: fedimintd/src/fedimintd.rs [hunk 1]
```diff
@@ -300,9 +304,7 @@ async fn run(
     // TODO: Fedimintd should use the config gen API
     // on each run we want to pass the currently passed password, so we need to
     // overwrite
-    if let Some(password) = opts.password {
```
Comment (dpc) [inline fedimintd/src/fedimintd.rs:303]: Do we have any protection against overwriting old password? Is it going to get overwritten every time now?

Comment (isaack-njama) [reply fedimintd/src/fedimintd.rs:303]: No protection to overwrite old password. Will need to check if a password exists before setting a new one.

> Do we have any protection against overwriting old password? Is it going to get overwritten every time now with a new random one?

File: fedimint-cli/src/lib.rs [hunk 2]
```diff
@@ -320,6 +321,15 @@ async fn load_or_generate_mnemonic(db: &Database) -> Result<Mnemonic, CliError>
     )
 }
 
+fn generate_random_password() -> String {
+    const PASSWORD_LENGTH: usize = 16;
+    thread_rng()
+        .sample_iter(&Alphanumeric)
```
Comment (elsirion) [inline fedimint-cli/src/lib.rs:327]: We should choose one case (upper or lower) and remove characters that look similar (`1`, `I`, `l`, …).

File: fedimintd/src/fedimintd.rs [hunk 3]
```diff
@@ -344,3 +353,12 @@ async fn spawn_metrics_server(
...
     Ok(())
 }
+
+fn generate_random_password() -> String {
+    const PASSWORD_LENGTH: usize = 16;
+    rand::thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(PASSWORD_LENGTH)
+        .map(char::from)
+        .collect()
```
Comment (elsirion) [inline fedimintd/src/fedimintd.rs:363]: We should not duplicate this code

**Review by elsirion — CHANGES_REQUESTED** [review]
I don't think this is the right approach. How does the guardian learn of the password if it's generated by the server?

I think the much easier solution is to generate a password in the admin UI and just not letting users set it there. If a technical user wants to set a custom password via `fedimint-cli` or the env var I'm ok with that.

**Comment by elsirion** [comment]
I think we should close this PR and just build it in the UI repo. See my reasoning above. Thoughts? @justinmoon @Kodylow

**Comment by Kodylow** [comment]
Agree

**Pull request closed without merging**